queries.sql

```diff
@@ -55,4 +55,91 @@ GROUP BY neutered;
+JOIN species ON animals.species_id = species.id WHERE species.name = 'Pokemon';
+
+SELECT animals.name AS animal_name, owners.fulL_name AS owner_name FROM owners 
+LEFT JOIN animals ON owners.id = animals.owners_id;
+
+SELECT species.name, COUNT(*) FROM animals 
+JOIN species ON species.id = animals.species_id GROUP BY species.name;
+
+SELECT animals.name AS animal_name, owners.full_name AS owner_name, species.name AS species_name FROM animals
+JOIN species ON species.id = animals.species_id JOIN owners ON owners.id = animals.owners_id WHERE species.name = 'Digimon' AND owners.full_name = 'Jennifer Orwell';
+
+SELECT * FROM animals
+JOIN owners ON animals.owners_id = owners.id 
+WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
+
+SELECT owners.full_name, COUNT(animals.name) AS num_animals FROM owners
+LEFT JOIN animals ON owners.id = animals.owners_id GROUP BY owners.full_name ORDER BY num_animals DESC;
+
+--visits
+
+--William Thatcher last visit
+
+SELECT vet.name AS vet_name, an.name AS animal_name
+  FROM visits v JOIN animals an ON an.id = v.animal_id
+  JOIN vets vet ON vet.id = v.vet_id
+  WHERE vet.name = 'William Tatcher'
+  ORDER BY v.date_of_visit DESC LIMIT 1;
+
+-- Stephanie Mendez  animal visits
+SELECT vet.name as vet_name, COUNT(date_of_visit) 
+  FROM visits v JOIN vets vet ON v.vet_id = vet.id
+  WHERE name = 'Stephanie Mendez' GROUP BY vet.name;
+
+  --all vets and specialties
+
+  SELECT 
+	vets.name,
+	species.name AS specialized_in
+FROM vets
+LEFT JOIN specializations ON specializations.vets_id = vets.id
+LEFT JOIN  species ON specializations.species_id = species.id;
+
+--Stephanie Mendez 2020-04-01-2020-08-30
+SELECT animals.name AS animal_name, visits.date_of_visit FROM animals
+JOIN visits ON visits.animals_id = animals.id
+JOIN vets ON vets.id = visits.vets_id
+WHERE vets.name = 'Stephanie Mendez' AND visits.date_of_visit >= '2020-04-01' AND visits.date_of_visit <= '2020-08-30';
+
+--most visits 
+SELECT an.name, COUNT(an.name) AS most_visits
+  FROM visits v JOIN animals an ON an.id = v.animal_id
+  GROUP BY an.name ORDER BY most_visits DESC LIMIT 1;
+
+  --first visit Maisy Smith
+  SELECT an.name AS animal, vet.name AS vet, v.date_of_visit
+  FROM visits v JOIN animals an ON an.id = v.animal_id
+  JOIN vets vet ON vet.id = v.vet_id
+  WHERE vet.name = 'Maisy Smith' ORDER BY v.date_of_visit ASC LIMIT 1;
+
+  --most recent visit
+  SELECT an.id AS animal_id, an.name AS animal, an.date_of_birth, vet.id AS vet_id, vet.name AS vet, vet.age AS vet_age, date_of_visit
+  FROM visits v JOIN animals an ON an.id = v.animal_id
+  JOIN vets vet ON vet.id = v.vet_id ORDER BY v.date_of_visit DESC
+  LIMIT 1;
+
+    --vet visit -no specialization
+    SELECT COUNT(*)
+FROM visits
+JOIN animals ON animals.id = visits.animals_id
+JOIN vets ON vets.id = visits.vets_id
+JOIN specializations ON specializations.vets_id = visits.vets_id
+WHERE animals.species_id != specializations.species_id;
```

Comment:
- Kindly check this query. This 👇 is what I got when I ran the query.

![](https://i.imgur.com/HaHtuAh.png)

From the data that wasinserted into the `visits` table, there should be 16 visits that were to a vet that did not specialize in that particular species.

This 👇 is an example of how the query could look like:

```
  SELECT count(*)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE animals.species_id NOT IN (SELECT species_id FROM specializations WHERE vet_id = vets.id)
```